GitUI/CommandsDialogs/FormPush.cs

```diff
@@ -682,29 +690,34 @@ private bool HandlePushOnExit(ref bool isError, FormProcess form)
             return (onRejectedPullAction ?? AppSettings.PullAction.None, forcePush);
         }
 
-        private void UpdateBranchDropDown()
+        private void _NO_TRANSLATE_Branch_Enter(object sender, EventArgs e)
+        {
+            if (_NO_TRANSLATE_Branch.Items.Count > 1)
+            {
+                return;
+            }
+
+            UpdateBranchDropDown(clear: false);
+            string curBranch = _NO_TRANSLATE_Branch.Text;
+            _NO_TRANSLATE_Branch.Items.RemoveAt(0);
+            _NO_TRANSLATE_Branch.Text = curBranch;
+        }
+
+        private void UpdateBranchDropDown(bool clear = true)
         {
             string curBranch = _NO_TRANSLATE_Branch.Text;
 
             _NO_TRANSLATE_Branch.DisplayMember = nameof(IGitRef.Name);
-            _NO_TRANSLATE_Branch.Items.Clear();
-            _NO_TRANSLATE_Branch.Items.Add(AllRefs);
-            _NO_TRANSLATE_Branch.Items.Add(HeadText);
 
-            if (string.IsNullOrEmpty(curBranch))
+            if (clear)
             {
-                Validates.NotNull(_currentBranchName);
-                curBranch = _currentBranchName;
-                if (curBranch.IndexOfAny("() ".ToCharArray()) != -1)
-                {
-                    curBranch = HeadText;
-                }
+                _NO_TRANSLATE_Branch.Items.Clear();
             }
 
-            foreach (IGitRef head in GetLocalBranches())
-            {
-                _NO_TRANSLATE_Branch.Items.Add(head);
-            }
+            _NO_TRANSLATE_Branch.Items.Add(AllRefs);
+            _NO_TRANSLATE_Branch.Items.Add(HeadText);
+
+            _NO_TRANSLATE_Branch.Items.AddRange(GetLocalBranches().ToArray());
```

Comment:
Consider SuspendLayout()/ResumeLayout(false); too (then the loop could have been kept, but AddRange is better)

Reply:
I will stick to the AddRange now that it is written 😉 
It's the first perf improvement I did and I though it give a good improvement but a later test show me that maybe it's negligible. It is at least compared to 2 other performance improvements made after. So I don't want to spend a lot of time on that. It's good enough for me.